staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -147,6 +147,11 @@ func (s *store) Create(ctx context.Context, key string, obj, out runtime.Object,
 	if version, err := s.versioner.ObjectResourceVersion(obj); err == nil && version != 0 {
 		return errors.New("resourceVersion should not be set on objects to be created")
 	}
+	// ensure that the serialized version does not contain any other fields that are not allowed
+	// to be stored at rest.
```

Comment:
Huh, so we don't want to store the resourceVersion. I guess that makes sense, since it gets overridden on the way out. A PrepareForStorage that empties resourceVersion and selfLink would be less likely to get called incorrectly and trigger unexpected behavior